sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/orc/OrcQuerySuite.scala

```diff
@@ -270,7 +270,10 @@ abstract class OrcQueryTest extends OrcTest {
   test("appending") {
     val data = (0 until 10).map(i => (i, i.toString))
     spark.createDataFrame(data).toDF("c1", "c2").createOrReplaceTempView("tmp")
-    withOrcTable(data, "t") {
+    // The test closure has side effect that changes table 't'. Running it without resetting
+    // the table data twice with vectorized=true and false will fail at the second time.
+    // Since it cares about inserting, we only do testing for vectorized=false.
```

Comment:
Originally, this test case cares about appending and reading back in the **vectorized** mode because the default value of `ORC_VECTORIZED_READER_ENABLED` is `true`. Can we keep the existing semantic?

Reply:
Ok. Because the test closure has side effect, I can't use `withOrcTable` for it. I slightly change the test but keep original semantic.

Reply:
+1